webapp/common/celery/celery.py

```diff
@@ -18,47 +18,106 @@
 
 import json
 from abc import ABC
+from typing import Callable
 
-from celery import Task
+from celery import Celery, Task, _state
 from flask import Flask
-from flask_celery import Celery
 from kombu.serialization import register
 
 from webapp.common.json import DefaultJSONEncoder
 
 
+class CeleryState:
+    """Remembers the configuration for the (celery, app) tuple. Modeled from SQLAlchemy."""
+
+    celery: Celery
+    app: Flask
+
+    def __init__(self, celery: Celery, app: Flask):
+        self.celery = celery
+        self.app = app
+
+
 class LogErrorsCelery(Celery):
+    """Celery extension for Flask applications.
+
+    Involves a hack to allow views and tests importing the celery instance from extensions.py to access the regular
+    Celery instance methods. This is done by subclassing celery.Celery and overwriting celery._state._register_app()
+    with a lambda/function that does nothing at all.
+
+    That way, on the first super() in this class' __init__(), all of the required instance objects are initialized, but
+    the Celery application is not registered. This class will be initialized in extensions.py but at that moment the
+    Flask application is not yet available.
+
+    Then, once the Flask application is available, this class' init_app() method will be called, with the Flask
+    application as an argument. init_app() will again call celery.Celery.__init__() but this time with the
+    celery._state._register_app() restored to its original functionality. in init_app() the actual Celery application is
+    initialized like normal.
+    """
+
+    original_register_app: Callable
+
+    def __init__(self):
+        """If app argument provided then initialize celery using application config values.
```

Comment:
```suggestion
        """If app argument is provided then initialize celery using application config values.
```

`initialize` is American English, and initialise is British English, in case it's important.